strings-csharp/CheckNumberInString/CheckNumberInString/ExtractNumber.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Globalization;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace CheckNumberInString
+{
+    public static partial class ExtractNumber
+    {
+        // Define valid numerical characters including digits, minus sign, and decimal point
+        private static readonly char[] validNumericalValues = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '-', '.' };
+
+        [GeneratedRegex(@"-?\d+(\.\d+)?", RegexOptions.Compiled)]
+        private static partial Regex NumberRegex();
+        public static string ExtractNumberUsingRegEx(string inputString)
+        {
+            var extractedNumbers = new List<double>();
+
+            var matches = NumberRegex().Matches(inputString);
+
+            foreach (Match match in matches)
```

Comment:
One more here that can change to `var`, unless you think it's less clear that way.

Reply:
This does not need to change since match.value is used in the loop and if used var code will through error.